aerleon/lib/naming.py

```diff
@@ -538,6 +540,34 @@ def GetServiceByProto(self, query: str, proto: str) -> List[str]:
                     services_set.add(parts[0])
         return sorted(services_set)
 
+    def GetFQDN(self, query: str) -> List[str]:
+        returnlist = []
+        data = query.split('#')
+        token = data[0].split()[0]
+        if token not in self.fqdn:
+            raise UndefinedAddressError(f'UNDEFINED: {token}')
+        for i in self.fqdn[token].items:
+            comment = ''
+            if i.find('#') > -1:
+                (name, comment) = i.split('#', 1)
+            else:
+                name = i
+
+            name = name.strip()
+            if self.token_re.match(name):
+                returnlist.extend(self.GetFQDN(name))
+            else:
+                try:
+                    fqdn = FQDN(name, token, comment)
+                    fqdn.text = comment.lstrip()
+                    fqdn.token = token
```

Comment:
Except for calling lstrip on comment, I think this is already handled in `FQDN.__init__` lines 15-18

Reply:
It was but I removed it because it will then be in line with the other function. This should likely be fixed in nacaddr and fqdn in a followup.